api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -223,6 +224,96 @@ public final Scope startScopedSpan() {
     return new ScopedSpanHandle(startSpan());
   }
 
+  /**
+   * Wraps a {@link Runnable} so that it executes with the {@code span} as the current {@code Span}
+   * and saves the rest of the Context, and ends the {@code Span} after the {@code runnable} is run.
+   *
+   * <p>This method saves the entire {@link io.grpc.Context} when wraps the {@code Runnable}.
+   *
+   * <p>This is equivalent with following code:
+   *
+   * <pre>{@code
+   * Span span = tracer.spanBuilder("MyRunnableSpan").startSpan();
+   * tracer.wrap(span, new Runnable() {
+   *     {@literal @}Override
+   *     public void run() {
+   *       try {
+   *         myRunnable.run();
+   *       } finally {
+   *         span.end();
+   *       }
+   *     }
+   * });
+   * }</pre>
+   *
+   * @param runnable the {@code Runnable} to run in the {@code Span}.
+   */
+  public final Runnable startSpanAndWrap(final Runnable runnable) {
+    final Span span = startSpan();
```

Comment:
it would be nice if you don't have to start a span in the builder. in brave we don't because there are a number of cases where you need a span visible before it is started (due to awkward callbacks ex http asyncclient)